test/cmdlineTests.sh

```diff
@@ -206,7 +206,7 @@ function test_solc_assembly_output()
     local expected_object="object \"object\" { code "${expected}" }"
 
     output=$(echo "${input}" | "$SOLC" - ${solc_args} 2>/dev/null)
-    empty=$(echo $output | sed -ne '/'"${expected_object}"'/p')
+    empty=$(echo "$output" | tr '\n' ' ' | tr -s ' ' | sed -ne "/${expected_object}/p")
```

Comment:
This one is interesting. `$output` looks like this in one of the cases:
```
Pretty printed source:
object "object" {
    code { let x := 0 }
}


Binary representation:
600050

Text representation:
    /* "<stdin>":16:22   */
  0x00
    /* "<stdin>":0:24   */
  pop
```

Not quoting it means that any `*` characters are evaluated as shell globs. And as you can see the output contains multi-line comments... The opening part (`/*`) expands to the content of your root directory while the closing one is replaced by the content of the current directory :) This means that part of the output is just garbage but the regex still finds the expected part because it is in there somewhere among that garbage :)

Quoting it fixes the problem but also breaks the regex because a convenient side-effect of not quoting it was that it got squashed into a single line with multiple spaces replaced by a single one. I added the `tr` commands to restore that behavior in a more explicit way.